lib/ddtrace/contrib/delayed_job/plugin.rb

```diff
@@ -9,7 +9,14 @@ class Plugin < Delayed::Plugin
         def self.instrument(job, &block)
           return block.call(job) unless tracer && tracer.enabled
 
-          tracer.trace(Ext::SPAN_JOB, service: configuration[:service_name], resource: job.name) do |span|
+           # If DelayedJob is used through ActiveJob, `job.name` is always "ActiveJob::QueueAdapters::DelayedJobAdapter::JobWrapper"
+          if job.payload_object.respond_to?(:job_data)
```

Comment:
The existence of this `job_data` method appears to be the key here to support ActiveJob.

Can you explain the differences between vanilla DelayedJobs and ones that come through ActiveJob? Specifically, what `job` is and how its configured? And could you also tell me a little more about the conditions when you expect each of these branches to be executed?

My thoughts behind those questions are that it would be best to avoid any specific configuration for ActiveJob in the DelayedJob integration, since it would introduce a kind of coupling via inversion of dependencies. However, what we have here does look pretty general purpose (which is promising); I'd just like to confirm it's not some kind of defacto coupling.

Reply:
`ActiveJob` is another abstraction, it’s the public API in front of any background jobs library. DelayedJob is one backend that executes the code that ActiveJob gives it. Rails uses ActiveJob to support a variety of queueing backends and DelayedJob is just one of them. 

I totally understand how creating an `ActiveJob` specific configuration creates an awkward inversion of dependencies. Knowing about the implementation details of ActiveJob isn't great.

I figured I’d dive down this avenue to see if y’all were amenable to supporting two different job payload objects. This solution already exists for [`Airbrake`](https://github.com/airbrake/airbrake/blob/master/lib/airbrake/delayed_job.rb#L14) for error catching, which is a good indicator that it's not too terrible of an idea — at least there's precedent for this kind of solution in other tools. 

Given that `ActiveJob` is the default framework for Rails 5+, I think it would be great to get this library patched up to work nicely with it for those folks (myself included) that are using `ActiveJob` with `DelayedJob`.

Reply:
I see, yeah. We're definitely wanting to add `ActiveJob` support to the library here in the future, and getting this working with DelayedJob would great.

Rails stuff can be a bit demanding of such special treatment, so I think I can appreciate that the change is at least fairly straightforward despite the introduction of coupling. I lean towards accepting it, but just wanted to make sure we're asking and answering the question "is this as simple and durable as it can be?"

Reply:
I think this can work as is. If we discover some better way of doing this in the future we can always change it, but for now it's good that we're getting some improved behavior out of this.